File: packages/shared/src/enums/UserFlag.ts
// Flag bits, always a power of 2 (2, 4, 8, ...)
export enum UserFlag {
  STAFF /*                */ = 1,
  TEMPORARILY_SUSPENDED /**/ = 2,
  PERMANENTLY_SUSPENDED /**/ = 4,
  VERIFIED /*             */ = 8,
  EARLY_USER /*           */ = 16
}

File: apps/backend/src/constants/user_flag.rs
use serde::{
    Deserialize,
    Serialize,
};

#[derive(Debug, Serialize, Deserialize, Copy, Clone)]
pub enum UserFlag {
    Staff = 1,
    TemporarilySuspended = 2,
    PermanentlySuspended = 4,
    Verified = 8,
    EarlyUser = 16,
}
